### ISSUE TITLE
Multi-Index in xarray data arrays

### ISSUE THREAD

**Issue by mpu-creare** [creare-com/podpac#301]
**Description**
I'm running the 010-retrieving-SMAP-Data.ipynb and I'm getting an error when I try to cache outputs. 
`NotImplementedError: variable 'lat_lon' is a MultiIndex, which cannot yet be serialized to netCDF files (https://github.com/pydata/xarray/issues/1077). Use reset_index() to convert MultiIndex levels into coordinate variables instead.`

**Steps to Reproduce**
```python
import podpac
time = podpac.crange('2018-05-15T00', '2018-05-19T00', '3,h')
lat_pts = [45.0, 45.0,  45.0, 30.0]
lon_pts = [-100.0, -70, -120.0, -100.0]
coords = podpac.Coordinates([(lat_pts, lon_pts), time], dims=['lat_lon', 'time'])
o = podpac.algorithm.Arange().eval(coords)
o.to_netcdf()
```

**Expected Behavior**
If I reset the index on o, then it works just fine. It looks like xarray has a new mechanism to define lat_lon paired coordinates? This is something we should investigate further... 

```python
>>> o
<xarray.UnitsDataArray (lat_lon: 4, time: 33)>
array([[nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan]])
Coordinates:
  * lat_lon  (lat_lon) MultiIndex
  - lat      (lat_lon) float64 45.0 45.0 45.0 30.0
  - lon      (lat_lon) float64 -100.0 -70.0 -120.0 -100.0
  * time     (time) datetime64[ns] 2018-05-15 2018-05-15T03:00:00 ... 2018-05-19
Attributes:
    layer_style:  <podpac.core.style.Style object at 0x00000211AC962B00>
    crs:          EPSG:4326
>>> o2 = o.reset_index('lat_lon')
>>> o2
<xarray.UnitsDataArray (lat_lon: 4, time: 33)>
array([[nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan]])
Coordinates:
  * time     (time) datetime64[ns] 2018-05-15 2018-05-15T03:00:00 ... 2018-05-19
    lat      (lat_lon) float64 45.0 45.0 45.0 30.0
    lon      (lat_lon) float64 -100.0 -70.0 -120.0 -100.0
Dimensions without coordinates: lat_lon
Attributes:
    layer_style:  <podpac.core.style.Style object at 0x00000211AC962B00>
    crs:          EPSG:4326
>>> o2.to_netcdf('test.nc')  # Works just fine

**Comment by jmilloy** [comment]
I looked into this a little bit.

We are using a multi-index for `StackedCoordinates`, but a multi-index cannot be serialized to netcdf. In fact, the [xarray docs](http://xarray.pydata.org/en/stable/reshaping.html#set-and-reset-index) say that the primary use for `reset_index` is serialization.

One solution here would be to drop the multi-indexes altogether. Another would be to `reset_index` before serializing, and then try to `set_index` when deserializing (loading).

The multi-index provides the ability to select (index using coordinates). For example, `o.sel(lat=45)` and `o.sel(lat_lon=(45, -70))` both work, but `o2.sel(lat=45)` and `on.sel(lat_lon=(35, -70))` don't work. The multi-index also easily provides the lat-lon pairs `o['lat_lon']` is meaningful, but `o2['lat_lon']` is not.

Note that we are handling the same thing *without the multi-index* in `DependentCoordinates`. Dependent coordinates are basically a generalization of stacked coordinates to nd-arrays, and they should probably be merged, see #268. It's not clear that we can do any meaningful indexing on nd stacked coordinates, so in this sense dropping the indexes makes sense.

@mpu-creare Should I try to preserve `sel` indexing when possible? Do you have any additional pro or cons related to keeping or removing the multi-indexes? #268 will take some time. The fastest bandaid would be to write a custom `UnitsDataArray.to_netcdf` method that calls `reset_index` when necessary, but removing multi-indexing from `StackedCoordinates` should be a quick enough fix anyways.

**Comment by mpu-creare** [comment]
I didn't realize that the multi-index continued to provide indexing functionality not otherwise available. In that case, I prefer to keep the multi-index as it is. 

For the latest release I used the `reset_index` call before serializing, so that works fine. I did not add the `set_index` when deserializing, so I'll create an issue for that. 

Thanks for looking into it @jmilloy. Closing!
